packages/base-driver/lib/basedriver/capabilities.ts

```diff
@@ -259,10 +268,12 @@ export function parseCaps<C extends Constraints>(
   let strippedRequiredCaps = stripAppiumPrefixes(requiredCaps);
   const strippedAllFirstMatchCaps: Capabilities<C>[] = allFirstMatchCaps.map(stripAppiumPrefixes);
 
-  // Validate the requiredCaps. But don't validate 'presence' because if that constraint fails on 'alwaysMatch' it could still pass on one of the 'firstMatch' keys
+  // Validate the requiredCaps. But don't validate 'presence' because if that constraint fails on 'alwaysMatch' it could still pass on one of the 'firstMatch' keys.
+  // Do not check deprecated property to reduce duplicated logs.
   if (shouldValidateCaps) {
     strippedRequiredCaps = validateCaps(strippedRequiredCaps, constraints, {
       skipPresenceConstraint: true,
+      skipDeprecatedCheck: true,
```

Comment:
I don't think this is going to help much.  We have at least 3 calls of validateCaps for different capability sets (e.g. parsed from firstMatch, alwaysMatch, etc). I expect some of them may have repeating cap names, so the message could be printed more than once, but it would be 'legit'. The problem there is that the validator itself is stateless, so we don't know from which context it's called and thus cannot keep any flags between calls to know which caps names we have already printed and which we did not in scope of the same driver.

Reply:
yes, correct. The root cause is the validator package and the method is stateless, so it can keep appium process wide state like https://github.com/appium/appium/pull/19988/commits/abb241f166624ba4e39e5913f3ffef4906752d8e but cannot control it per session level.

I'm currently ok to close this and leave duplication messages (it should be better than the previous 3 lines bug).